airlock/workspace_api.py

```diff
@@ -0,0 +1,53 @@
+import dataclasses
+import pathlib
+
+from django.conf import settings
+from django.urls import reverse
+
+
+@dataclasses.dataclass(frozen=True)
+class PathItem:
+    relpath: pathlib.Path
+
+    @classmethod
+    def from_relative_path(cls, path: str | pathlib.Path):
+        return cls.from_absolute_path(settings.WORKSPACE_DIR / path)
+
+    @classmethod
+    def from_absolute_path(cls, path: pathlib.Path):
+        return cls(path.resolve().relative_to(settings.WORKSPACE_DIR))
+
+    def absolute_path(self):
+        return settings.WORKSPACE_DIR / self.relpath
+
+    def exists(self):
+        return self.absolute_path().exists()
+
+    def is_directory(self):
+        return self.absolute_path().is_dir()
+
+    def name(self):
+        return self.relpath.name
+
+    def url(self):
+        suffix = "/" if self.is_directory() else ""
+        return reverse("file_browser", kwargs={"path": f"{self.relpath}{suffix}"})
+
+    def parent(self):
+        if self.relpath.parents:
+            return PathItem(self.relpath.parent)
+
+    def children(self):
```

Comment:
Makes me slightly uneasy to see this on something that might represent a file.

Reply:
Yeah, I felt similarly. And also having a `contents()` method on a class which might represent a directory. But having an abstract base class with two subclasses felt a bit heavy-handedly OO (at least at this stage). And I couldn't think of what concrete advantages it would give us given that we're not actually type checking. Obviously both methods raise errors if you call them on the wrong sort of thing, which in dynamic terms isn't much different from them just not being there.

Reply:
Fine.